### DISCUSSION TITLE
Opendeck L and analog inputs

### DISCUSSION THREAD

**Discussion by pmjcontroller1** [shanteacontrols/OpenDeck#228] [Q&A] [answered]
Hi, @paradajz 
I'm testing a opendeck L. All digital inputs and outputs work fine, but not the analogs. Of course, I activate it in the setup.

I have tried to inject analog signal directly to the mux_common to get at least any response, but it seems like dead.

Besides, I don't have any signal in the control pins with the oscilloscope. 

Any idea?

**Comment by pmjcontroller1** [comment]
Ok...the analog pin of the rpi2040 works when I configure en the yml as native.
The problem is focused in the muxonmux setup. I don't know what's happens. Any pin involved has activity

**Comment by pmjcontroller1** [comment]
![16928103882688295498777675433160](https://github.com/shanteacontrols/OpenDeck/assets/118101014/268e9837-038a-4dc4-b9bb-2e33985e75f2)
This is L board

**Comment by paradajz** [comment]
Did you build that board yourself?

**Reply by pmjcontroller1** [reply]
we build it with your design version 3.1

**Reply by paradajz** [reply]
Disregard my previous comment - I actually do use 5V signaling for the multiplexers on the L board. I'm not really sure what's wrong.

**Reply by pmjcontroller1** [reply]
I use the74hct125 to adapt the voltages.
It is very strange. I have made a separately setup for rpi pico board with only one multiplexer in a protoboard to check the program in a  micro separately and the micro doesn't do any. Any signals for control and any value in the analog pin.
If I put the analog like "native" it works.

**Comment by pmjcontroller1** [comment]
Good morning @paradajz .
any suggestions for this problem? before we had done tests with a pi and it worked. We decided to include your systems in our controllers, and we manufacture your L plates. Once they work, we plan to make you a donation, since we would include your project in our systems. Could you please help us. We think it must be something in the code, so we explained it to you in the previous message. Any need for verification that you need we would do. thank you

**Comment by pmjcontroller1** [comment]
![IMG20230829132757.jpg](https://github.com/shanteacontrols/OpenDeck/assets/118101014/b76ceaca-bc4f-4a86-a9bc-1f8d13e8033a)

**Comment by pmjcontroller1** [comment]
Hi @paradajz ,
I have made some test and my final conclusion is the control mux 4051 doens't work properly.
I have connected the common pin of the first 4067 to the pin 3 of the 5051 (common pin on 4051 --> gpio28 on rp2040) directly, configured the yaml file according and the board works fine with this configuration. 
I think that the control mux pins don't work. 
Is it possible a software bug?
Thks!!!

**Comment by paradajz** [comment]
Can you try loading this firmware?
https://github.com/shanteacontrols/OpenDeck/releases/tag/v7.4.0

This was the first release that brought support for that board.

**Reply by pmjcontroller1** [reply]
Hi,
yes!!! this firmware works fine...

Where is the differences between version?

**Reply by paradajz** [reply]
Ouf, then I have a bug. Can you help me on this? Can you go through all the firmware releases in order from v7.4.0 and check in which firmware it doesn't work anymore? Sadly I don't have any L board right here to test.

**Reply by pmjcontroller1** [reply]
The firmware works in all 7.4.x versions. It is from 7.5.x when fails.

**Reply by paradajz** [reply]
Thanks. Found the bug. Can you try this firmware? It's the latest one with the fix applied.

[merged.zip](https://github.com/shanteacontrols/OpenDeck/files/12497114/merged.zip)

**Reply by pmjcontroller1** [reply]
Works!!!

**Reply by paradajz** [reply, marked as answer]
Thanks for getting back. New official release is up:
https://github.com/shanteacontrols/OpenDeck/releases/tag/v7.5.4
